src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -313,13 +322,13 @@ public virtual ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int a
                 }
             }
 
-            namespaceOrTypeMembers = scope.GetTypeMembers(emittedTypeName.TypeName);
+            namespaceOrTypeMembers = scope.GetTypeMembers(emittedTypeName.TypeNameMemory);
 
             foreach (var named in namespaceOrTypeMembers)
             {
                 if (!named.MangleName &&
                     (forcedArity == -1 || forcedArity == named.Arity) &&
-                    named.MetadataName == emittedTypeName.TypeName)
+                    named.MetadataName.AsSpan().SequenceEqual(emittedTypeName.TypeNameMemory.Span))
```

Comment:
We could consider some extensino methods to help compare strings to ROM, and whatnot.